packages/material-ui/src/Menu/Menu.js

```diff
@@ -44,23 +50,50 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     disableAutoFocusItem = false,
     MenuListProps = {},
     onClose,
+    onEnter,
     onEntering,
+    onEntered,
     open,
     PaperProps = {},
+    setParentLastEnteredItemIndex,
     PopoverClasses,
     transitionDuration = 'auto',
+    isSubMenu = false,
     variant = 'selectedMenu',
     ...other
   } = props;
   const theme = useTheme();
 
+  const [lastEnteredItemIndex, setLastEnteredItemIndex] = React.useState(null);
```

Comment:
This name is very descriptive but can we shorten it at all? What about if we remove the "Entered": lastItemIndex?